src/dictionaries.cpp

```diff
@@ -18,17 +18,27 @@ uint32_t BasicDictionary::code(const std::string& name) const
 
 void BasicDictionary::add(uint32_t code, const std::string& name)
 {
-    m_rightDict.emplace(code, name);
+    for (const auto& entry: m_rightDict)
+        if (entry.second == name)
+            throw RadProto::Exception(RadProto::Error::suchAttributeNameAlreadyExists, name + "- add");
```

Comment:
Taking into account my comment above, here you should provide the full message, not just a suffix.

Reply:
Also, what if we add the same name with the same code? As we discussed it earlier, this should be a no-op.